### PR TITLE
[2.9-python3-acceptancetests] Force python3 when running acceptance tests

### PR DESCRIPTION
**Pull request by achilleasa**
This PR applies a few changes to make some of the tests suitable for running using python3. More specifically:
- decode byte slices prior to testing regexes
- specify 'rb' access mode when reading binary files (python3 attempts to decode contents as utf-8 and explodes)

The PR targets a special branch `2.9-python3-acceptancetests` which allows us to land this set of changes, have the CI run all acceptance tests and fix the ones that break due to python2/python3 incompatibilities in a followup PR.

## QA steps

- make install from this branch
- install the latest 2.9 rc snap
- ./assess_upgrade.py lxd `which juju` /tmp/art nw-upgrade-juju-amd64-lxd --stable-juju-bin /snap/juju/current/bin/juju

### CODE REVIEW

File: acceptancetests/jujupy/client.py
Comment (achilleasa) [inline acceptancetests/jujupy/client.py:568]: When running the `assess_upgrade` test, this func gets called with **both** a bytes version of `version_string` as well as a string one.

File: acceptancetests/assess
Comment (petevg) [inline acceptancetests/assess:1]: Typo! This should be python3.

**Review by petevg — CHANGES_REQUESTED** [review]
Caught a small typo. Everything else looks fine, or at least as expected, if the plan is to switch over to Python3, and make fixes as you catch failures.

**Comment by petevg** [comment]
As a general note, it might be interesting to run these files through a linter. You'd probably get lots of noise, but it should also catch a lot of the most straightforwardly broken things, like missing parens after print and such.

File: acceptancetests/assess
Comment (achilleasa) [reply acceptancetests/assess:1]: Whoops... Forgot to properly terminate my find and replace regex :D

**Comment by achilleasa** [comment]
> As a general note, it might be interesting to run these files through a linter. You'd probably get lots of noise, but it should also catch a lot of the most straightforwardly broken things, like missing parens after print and such.

There are quite a few linter warnings (most are of the too-long-lines kind though) and I am trying to fix as many as I see (that's how I fixed the missing parenthesis one) but I am not generally running the linter against all files; just the ones open in my vim buffers. I guess since I will spend a bit of time on porting these I might as well fix all warnings.

File: acceptancetests/assess_juju_output.py
Comment (SimonRichardson) [inline acceptancetests/assess_juju_output.py:109]: I'm guessing this actually solves a test?

File: acceptancetests/assess_model_migration_versions.py
Comment (SimonRichardson) [inline acceptancetests/assess_model_migration_versions.py:70]: Ah, this is because it's using the wrong client?

File: acceptancetests/assess_wallet.py
Comment (SimonRichardson) [inline acceptancetests/assess_wallet.py:126]: Is this correct, does `JujuAssertionError` implement `__str__` correctly?

**Review by SimonRichardson — APPROVED** [review]
I have a few questions, but not many changes...

File: acceptancetests/assess_model_migration_versions.py
Comment (achilleasa) [reply acceptancetests/assess_model_migration_versions.py:70]: `migrate_model_to_controller` signature is `(model, src_client, dst_client)` (no defaults!). And both this and the invocation below it used 2 args so I changed the args based on the comment above it.

File: acceptancetests/assess_wallet.py
Comment (achilleasa) [reply acceptancetests/assess_wallet.py:126]: JujuAssertionError inherits from the built-in AssertionError (which I believe should implement that). It has no other visible fields in the docs so I hope it does :D

File: acceptancetests/assess_juju_output.py
Comment (achilleasa) [reply acceptancetests/assess_juju_output.py:109]: Not really sure what it does but given that:
a) it's missing
b) the bits below check specific status fields anyway

it seemed like the best approach would be to get rid of it.

**Comment by achilleasa** [comment]
@petevg As you are off today and I won't be able to land (even though Simon already approved the extra bits) until you also approve the changes I will go ahead and click the dismiss review button so I can land this.

**Comment by achilleasa** [comment]
$$merge$$

**Comment by achilleasa** [comment]
$$merge$$ (transient error; PR only changes acceptance tests)

**Pull request merged**
